src/table/townname.h

```diff
@@ -9,6 +9,10 @@
 
 #include "../core/enum_type.hpp"
 
+#if defined(__GNUC__)
+#pragma GCC diagnostic ignored "-Wunused-variable"
```

Comment:
still missing tabs :)

Current CI builds don't show an "unused variable" anywhere. Is this still current? Maybe add a comment about which variable it was?

Reply:
I definitely saw it for the release builds on azure. Might just a spurious warning from older compilers?

Reply:
Are you now really asking me? :D

https://github.com/TrueBrain/OpenTTD/actions/runs/423369687

A recent run with the compilers we still support. None of them showed any GCC warning about unused variable.

https://github.com/TrueBrain/OpenTTD/actions/runs/419515685

16.04 reported it .. but I do not think we should be fixing GCC bugs in this broad fashion. If we do it, we should clamp it to the GCC version that is triggering the problem, I think.